src/main/java/use_case/registration/RegistrationDataAccessInterface.java

```diff
@@ -0,0 +1,8 @@
+package use_case.registration;
+
+import entity.User;
+
+public interface RegistrationDataAccessInterface {
+    void saveUser(User user);
+    User getUserWithUsername(String username);
```

Comment:
The only thing `RegistrationDataAccessInterface` needs to do is to read a `User` object and store it to database. It does not need a `getUser` method.

Reply:
1. add some @Javadoc
2. The interface should first check if the user already exists, if so, let a dialogue pop up; if not, it should be able to save the user.